### PR TITLE
Allow Admin user to delete Collections

### PR DESCRIPTION
**Pull request by manuel-delvillar**
- If you have admin role and enable roles is active, the button for delete a collection should be visible
Screenshot 2023-01-23 at 16 03 13

- If you don't have a role admin or if enable roles is not active, the button for delete should not be visible.
Screenshot 2023-01-23 at 16 05 21

- You need to confirm, you really want to delete a collection

Screenshot 2023-01-23 at 16 07 35

- After delete a collection you are redirected to the list of collections.
- After delete a collection no longer appear in an RSD detail.

### CODE REVIEW

File: api/src/main/kotlin/edu/wgu/osmt/collection/CollectionRepository.kt [hunk 1]
```diff
@@ -197,6 +206,30 @@ class CollectionRepositoryImpl @Autowired constructor(
         return daoObject
     }
 
+    override fun remove(uuid: String): ApiBatchResult {
+
+        val collectionFound = findByUUID(uuid)
+
+        if (collectionFound != null) {
+            collectionSkillsTable.deleteWhere { collectionSkillsTable.collectionId eq collectionFound.id }
+            collectionFound?.let { collectionEsRepo.delete(it.toDoc()) }
```
Comment (JohnKallies) [inline api/src/main/kotlin/edu/wgu/osmt/collection/CollectionRepository.kt:215]: Should these 2 deletes (db repo and ES repo) be in a transaction? What does it cost if yes? What's the risk if no?

File: api/src/main/kotlin/edu/wgu/osmt/task/TaskMessageService.kt [hunk 2]
```diff
@@ -43,6 +43,7 @@ class TaskMessageService {
         const val publishSkills = "batch-publish-skills"
         const val updateCollectionSkills = "update-collection-skills"
         const val skillsForCollectionCsv = "collection-skills-csv-process"
+        const val removeCollectionSkills = "remove-collection-skills"
```
Comment (JohnKallies) [inline api/src/main/kotlin/edu/wgu/osmt/task/TaskMessageService.kt:46]: Is this for removing collection skills? Or more generalized to removing a collection? I'm not passive aggressively saying what I think, instead I'm genuinely asking your opinion.

File: api/src/test/kotlin/edu/wgu/osmt/collection/CollectionRepositoryTest.kt [hunk 3]
```diff
@@ -16,6 +16,7 @@ import edu.wgu.osmt.richskill.RsdUpdateObject
 import edu.wgu.osmt.task.PublishTask
 import edu.wgu.osmt.task.UpdateCollectionSkillsTask
 import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Disabled
```
Comment (JohnKallies) [inline api/src/test/kotlin/edu/wgu/osmt/collection/CollectionRepositoryTest.kt:19]: Looks like you are still working in this file. When done, would you please ensure that only related changes are included?

File: api/src/test/kotlin/edu/wgu/osmt/collection/CollectionRepositoryTest.kt [hunk 4]
```diff
@@ -189,4 +191,28 @@ class CollectionRepositoryTest: SpringTest(), BaseDockerizedTest, HasDatabaseRes
         assertThat(batchResult?.modifiedCount).isEqualTo(skillCount*3)
     }
 
+    @Test
+    fun `remove finds and succesfully removes an existing collection`() {
+        // Arrange
+        val collection = collectionRepository.create(UUID.randomUUID().toString(), userString)!!.toModel()
+
```
Comment (JohnKallies) [inline api/src/test/kotlin/edu/wgu/osmt/collection/CollectionRepositoryTest.kt:198]: Can we please add some RSDs to test the delete from the relationship table?

File: api/src/main/kotlin/edu/wgu/osmt/task/TaskMessageService.kt (hunk 2 above)
Comment (jchavez137) [reply api/src/main/kotlin/edu/wgu/osmt/task/TaskMessageService.kt:46]: you are totally right, it should only be remove collection, it was a copy/paste name routine.

File: api/src/main/kotlin/edu/wgu/osmt/security/SecurityConfig.kt [hunk 5]
```diff
@@ -29,8 +30,7 @@ import org.springframework.beans.factory.annotation.Autowired
 import org.springframework.context.annotation.Bean
 import org.springframework.context.annotation.Configuration
 import org.springframework.context.annotation.Profile
-import org.springframework.http.HttpMethod.GET
-import org.springframework.http.HttpMethod.POST
+import org.springframework.http.HttpMethod.*
```
[the remark this answers is not in the document]
Comment (jchavez137) [reply api/src/main/kotlin/edu/wgu/osmt/security/SecurityConfig.kt:33]: (facepalm)

File: api/src/main/kotlin/edu/wgu/osmt/collection/CollectionRepository.kt (hunk 1 above)
Comment (jchavez137) [reply api/src/main/kotlin/edu/wgu/osmt/collection/CollectionRepository.kt]: changed logic here

File: api/src/main/kotlin/edu/wgu/osmt/security/SecurityConfig.kt (hunk 5 above)
[the remark this answers is not in the document]
Comment (jchavez137) [reply api/src/main/kotlin/edu/wgu/osmt/security/SecurityConfig.kt:33]: reverted

File: api/src/main/kotlin/edu/wgu/osmt/task/TaskMessageService.kt (hunk 2 above)
Comment (jchavez137) [reply api/src/main/kotlin/edu/wgu/osmt/task/TaskMessageService.kt]: changed

File: api/src/test/kotlin/edu/wgu/osmt/collection/CollectionRepositoryTest.kt (hunk 4 above)
Comment (jchavez137) [reply api/src/test/kotlin/edu/wgu/osmt/collection/CollectionRepositoryTest.kt:200]: added assertion for RSD table changes

File: ui/src/app/collection/detail/manage-collection.component.ts [hunk 6]
```diff
@@ -276,6 +302,7 @@ export class ManageCollectionComponent extends SkillsListComponent implements On
   }
 
   submitCollectionStatusChange(newStatus: PublishStatus, verb: string): void {
+    console.log("submit collection status change")
```
Comment (JohnKallies) [inline ui/src/app/collection/detail/manage-collection.component.ts:305]: remove

File: ui/src/app/collection/detail/manage-collection.component.ts [hunk 7]
```diff
@@ -315,7 +342,8 @@ export class ManageCollectionComponent extends SkillsListComponent implements On
     const count = (this.apiSearch?.uuids?.length ?? 0)
 
     if (count > 1 || this.selectAllChecked) {
-      this.showingMultipleConfirm = true
+      // this.showingMultipleConfirm = true
```
Comment (JohnKallies) [inline ui/src/app/collection/detail/manage-collection.component.ts:345]: remove? or use?

File: ui/src/app/collection/detail/manage-collection.component.ts [hunk 8]
```diff
@@ -340,13 +368,15 @@ export class ManageCollectionComponent extends SkillsListComponent implements On
 
   handleClickConfirmMulti(): boolean {
     this.submitSkillRemoval(this.apiSearch)
-    this.showingMultipleConfirm = false
+    // this.showingMultipleConfirm = false
```
Comment (JohnKallies) [inline ui/src/app/collection/detail/manage-collection.component.ts:371]: again, please

File: ui/src/app/collection/detail/manage-collection.component.ts [hunk 9]
```diff
@@ -340,13 +368,15 @@ export class ManageCollectionComponent extends SkillsListComponent implements On
...
+    this.template = "default"
     this.apiSearch = undefined
     return false
   }
 
   handleClickCancel(): boolean {
-    this.showingMultipleConfirm = false
+    // this.showingMultipleConfirm = false
```
Comment (JohnKallies) [inline ui/src/app/collection/detail/manage-collection.component.ts:378]: :)

**Review by JohnKallies — APPROVED** [review]

**Pull request merged**